### PR TITLE
CSPL-1302: Bias-language removal Phase 1 [Comments & Docs]

### PR DESCRIPTION
**Pull request by mgaldino-splunk**
Removing Bias Language - Phase I.

In this PR we are removing bias language from **Comments** and **Documentation**.  There is no change in the actual Operator code, therefore we do not anticipate the need for QA in this PR.

This is only the first PR to remove bias language. More details will be shared with the team shortly.

**Progress status:**

**Before PR:**
  "terms_found": true,
  "total_lines_matched": 1987,
  "total_words_matched": 2657,
  "total_files_matched": 135

**After PR:**
  "terms_found": true,
  "total_lines_matched": 1806,
  "total_words_matched": 2417,
  "total_files_matched": 132

### CODE REVIEW

File: docs/Examples.md [hunk 1]
```diff
@@ -725,7 +725,7 @@ There are two ways to configure `pass4Symmkey` with an External LM:
   ```
 - Setup the above decrypted plain-text [`pass4Symmkey`](PasswordManagement.md#pass4Symmkey) in the global secret object(Note: The `pass4Symmkey` would be stored in a base64 encoded format). For details see [updating global kubernetes secret object](#updating-global-kubernetes-secret-object)
 
-### Configuring license_master_url:
+### Configuring license_manager_url:
```
Comment (jryb) [inline docs/Examples.md:728]: Probably should leave this until this env variable name changes.

File: pkg/apis/enterprise/v2/indexercluster_types.go [hunk 2]
```diff
@@ -106,7 +106,7 @@ type IndexerClusterStatus struct {
 // +kubebuilder:subresource:scale:specpath=.spec.replicas,statuspath=.status.replicas,selectorpath=.status.selector
 // +kubebuilder:resource:path=indexerclusters,scope=Namespaced,shortName=idc;idxc
 // +kubebuilder:printcolumn:name="Phase",type="string",JSONPath=".status.phase",description="Status of indexer cluster"
-// +kubebuilder:printcolumn:name="Master",type="string",JSONPath=".status.clusterMasterPhase",description="Status of cluster master"
+// +kubebuilder:printcolumn:name="Manager",type="string",JSONPath=".status.clusterMasterPhase",description="Status of cluster manager"
```
Comment (jryb) [inline pkg/apis/enterprise/v2/indexercluster_types.go:109]: This one I'm not sure about.  Whats `name` in this context? Probably ok though if all automation tests pass.

File: pkg/splunk/client/enterprise.go [hunk 3]
```diff
@@ -424,23 +424,23 @@ func (c *SplunkClient) GetClusterMasterInfo() (*ClusterMasterInfo, error) {
 }
 
 // IndexerClusterPeerInfo represents the status of a indexer cluster peer.
-// See https://docs.splunk.com/Documentation/Splunk/latest/RESTREF/RESTcluster#cluster.2Fslave.2Finfo
+// See https://docs.splunk.com/Documentation/Splunk/latest/RESTREF/RESTcluster#cluster.2Fpeer.2Finfo
```
Comment (jryb) [inline pkg/splunk/client/enterprise.go:427]: Just a comment to make sure these url/anchors still work.

File: pkg/splunk/client/enterprise.go [hunk 4]
```diff
@@ -452,7 +452,7 @@ type IndexerClusterPeerInfo struct {
 
 // GetIndexerClusterPeerInfo queries info from a indexer cluster peer.
 // You can use this on any peer in an indexer cluster.
-// See https://docs.splunk.com/Documentation/Splunk/latest/RESTREF/RESTcluster#cluster.2Fslave.2Finfo
+// See https://docs.splunk.com/Documentation/Splunk/latest/RESTREF/RESTcluster#cluster.2Fpeer.2Finfo
```
Comment (jryb) [inline pkg/splunk/client/enterprise.go:455]: Same as above.

File: pkg/splunk/client/enterprise.go (hunk 3 above)
Comment (mgaldino-splunk) [reply pkg/splunk/client/enterprise.go:427]: Thanks Jeff. Double checked they work as the documentation is already using the word "peer".

File: pkg/splunk/client/enterprise.go (hunk 4 above)
Comment (mgaldino-splunk) [reply pkg/splunk/client/enterprise.go:455]: Validated they work. Thanks Jeff.

File: docs/Examples.md (hunk 1 above)
Comment (mgaldino-splunk) [reply docs/Examples.md:728]: That's a good point. We can roll them back for now.

File: pkg/apis/enterprise/v2/indexercluster_types.go (hunk 2 above)
Comment (mgaldino-splunk) [reply pkg/apis/enterprise/v2/indexercluster_types.go:109]: This code is all commented so I believe it's a legacy no longer used.  I thought even better would be to removing it altogether. But not sure if this serves as a reference.

Comment (jryb) [reply pkg/apis/enterprise/v2/indexercluster_types.go:109]: I don't think it's old code.  I think these are `kubebuilder` tags for when the CRDs are generated.  Though looking at the docs, this name change is likely fine: https://book.kubebuilder.io/reference/generating-crd.html#additional-printer-columns

Comment (mgaldino-splunk) [reply pkg/apis/enterprise/v2/indexercluster_types.go:109]: Nice information Jeff. I think we can roll it back for now, and leave this change for when we deal with the CRDs.

**Review by gaurav-splunk — APPROVED** [review]

**Review by jryb — APPROVED** [review]

File: docs/Examples.md (hunk 1 above)
Comment (mgaldino-splunk) [reply docs/Examples.md]: Done.

File: pkg/apis/enterprise/v2/indexercluster_types.go (hunk 2 above)
Comment (mgaldino-splunk) [reply pkg/apis/enterprise/v2/indexercluster_types.go]: rolled back.

File: pkg/splunk/client/enterprise.go [hunk 5]
```diff
@@ -470,7 +470,7 @@ func (c *SplunkClient) GetIndexerClusterPeerInfo() (*IndexerClusterPeerInfo, err
 	return &apiResponse.Entry[0].Content, nil
 }
 
-// ClusterMasterPeerInfo represents the status of a indexer cluster peer (cluster master endpoint).
+// ClusterMasterPeerInfo represents the status of a indexer cluster peer (cluster manager endpoint).
```
Comment (kashok-splunk) [inline pkg/splunk/client/enterprise.go:473]: Do we want to take care of these struct/function names too?

File: pkg/splunk/client/enterprise.go [hunk 6]
```diff
@@ -572,8 +572,8 @@ type ClusterMasterPeerInfo struct {
 	} `json:"status_counter"`
 }
 
-// GetClusterMasterPeers queries the cluster master for info about indexer cluster peers.
-// You can only use this on a cluster master.
+// GetClusterMasterPeers queries the cluster manager for info about indexer cluster peers.
```
Comment (kashok-splunk) [inline pkg/splunk/client/enterprise.go:575]: one more here

File: pkg/splunk/enterprise/licensemaster.go [hunk 7]
```diff
@@ -123,7 +123,7 @@ func ApplyLicenseMaster(client splcommon.ControllerClient, cr *enterpriseApi.Lic
 	return result, nil
 }
 
-// getLicenseMasterStatefulSet returns a Kubernetes StatefulSet object for a Splunk Enterprise license master.
+// getLicenseMasterStatefulSet returns a Kubernetes StatefulSet object for a Splunk Enterprise license manager.
```
Comment (kashok-splunk) [inline pkg/splunk/enterprise/licensemaster.go:126]: same here

**Review by kashok-splunk — COMMENTED** [review]
Do we need to take care of function/struct/variable names that contain "Master" here? Or that is not within the scope of this PR?
And in https://github.com/splunk/splunk-operator/blob/develop/pkg/splunk/enterprise/types.go#L49 maybe we should change the names of SplunkClusterMaster & SplunkLicenseMaster?

File: pkg/splunk/client/enterprise.go (hunk 5 above)
Comment (mgaldino-splunk) [reply pkg/splunk/client/enterprise.go:473]: Great point Kriti. These cases will be taken care of in future PRs. The reason is that the Functions/Structs themselves will have their name changed. A few more PRs coming in the future :D. For now, we'll keep these out of scope.

File: pkg/splunk/client/enterprise.go (hunk 6 above)
Comment (mgaldino-splunk) [reply pkg/splunk/client/enterprise.go:575]: same as above.

File: pkg/splunk/enterprise/licensemaster.go (hunk 7 above)
Comment (mgaldino-splunk) [reply pkg/splunk/enterprise/licensemaster.go:126]: same as above

File: docs/Examples.md [hunk 8]
```diff
@@ -459,8 +459,8 @@ spec:
 ```
 
 
-### Example: Cluster Master
-In the ClusterMaster example, app3 and app4 are installed on any indexer instances that are managed by the cluster master. App5 and app6 are installed locally on the ClusterMaster instance.
+### Example: Cluster Manager
+In the ClusterMaster example, app3 and app4 are installed on any indexer instances that are managed by the cluster manager. App5 and app6 are installed locally on the ClusterMaster instance.
```
Comment (eddikay) [inline docs/Examples.md:463]: Let's just use "In this example, app3...."

**Review by eddikay — CHANGES_REQUESTED** [review]
Reviewing the .md files, the use of straight replacement text looks fine, except when there's a mismatch of terms due to the use of the ClusterMaster CR title in close proximity to a changed role name. I've commented on the one that stuck out.

File: docs/Examples.md (hunk 8 above)
Comment (mgaldino-splunk) [reply docs/Examples.md:463]: Hi @eddikay.  If I understood correctly you're suggesting to replace:

`In the ClusterMaster example, app3 ...`

with 

`In this example, app3 ...`

Comment (eddikay) [reply docs/Examples.md:463]: Yes! Thank you :)

Comment (mgaldino-splunk) [reply docs/Examples.md]: done.

**Review by kashok-splunk — APPROVED** [review]

File: pkg/splunk/client/enterprise.go [hunk 9]
```diff
@@ -470,7 +470,7 @@ func (c *SplunkClient) GetIndexerClusterPeerInfo() (*IndexerClusterPeerInfo, err
 	return &apiResponse.Entry[0].Content, nil
 }
 
-// ClusterMasterPeerInfo represents the status of a indexer cluster peer (cluster master endpoint).
+// ClusterMasterPeerInfo represents the status of a indexer cluster peer (cluster manager endpoint).
 // See https://docs.splunk.com/Documentation/Splunk/latest/RESTREF/RESTcluster#cluster.2Fmaster.2Fpeers
```
Comment (smohan-splunk) [inline pkg/splunk/client/enterprise.go:474]: Update this URL too?
https://docs.splunk.com/Documentation/Splunk/latest/RESTREF/RESTcluster#cluster.2Fmanager.2Fpeers

**Review by smohan-splunk — APPROVED** [review]

File: pkg/splunk/client/enterprise.go (hunk 9 above)
Comment (mgaldino-splunk) [reply pkg/splunk/client/enterprise.go]: done

**Review by eddikay — APPROVED** [review]

**Pull request merged**